extension/observer/kafkatopicsobserver/extension.go

```diff
@@ -24,35 +26,116 @@ var (
 
 type kafkaTopicsObserver struct {
 	*observer.EndpointsWatcher
-	logger *zap.Logger
-	config *Config
-	cancel func()
-	once   *sync.Once
-	_      context.Context
+	logger     *zap.Logger
+	config     *Config
+	cancel     func()
+	once       *sync.Once
+	ctx        context.Context
```

Comment:
I would strongly advise against storing context since it can be invalid at any point, and it can also encourage some non go idiomatic behaviour.